File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.py
import sys

t = int(sys.stdin.readline().rstrip())
n = []


for _ in range(t):
    n.append(int(sys.stdin.readline().rstrip()))

n_sorted = sorted(n)
zero = [0] * 100
one = [0] * 100
zero[0] = 1
one[1] = 1

if n_sorted[len(n_sorted) - 1] > 2:
    for i in range(2, n_sorted[len(n_sorted) - 1] + 1):
        zero[i] = zero[i - 2] + zero[i - 1]
        one[i] = one[i - 2] + one[i - 1]

for i in n:
    print(zero[i], end=' ')
    print(one[i], end='\n')

File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.java
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int[] zero = new int[41];
        int[] one = new int[41];
        zero[0] = 1;
        zero[1] = 0;
        one[0] = 0;
        one[1] = 1;
        for (int i = 2; i < 41; i++) {
            zero[i] = zero[i - 1] + zero[i - 2];
            one[i] = one[i - 1] + one[i - 2];
        }

        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n; i++) {
            int t = Integer.parseInt(br.readLine());
            bw.write(zero[t] + " " + one[t] + "\n");
        }

        bw.flush();
        bw.close();
    }
}